tests/Equinox.Cosmos.Integration/JsonConverterTests.fs

```diff
@@ -18,20 +19,29 @@ let defaultOptions = FsCodec.SystemTextJson.Options.Create()
 type Base64ZipUtf8Tests() =
     let eventCodec = FsCodec.SystemTextJson.Codec.Create<Union>(defaultOptions)
 
-    [<Fact>]
-    let ``serializes, achieving compression`` () =
-        let encoded = eventCodec.Encode(None,A { embed = String('x',5000) })
+    [<Property>]
+    let ``Can read compressed`` value =
+        let hasNulls =
+            match value with
+            | A x | B x when obj.ReferenceEquals(null, x) -> true
+            | A { embed = x } | B { embed = x } -> obj.ReferenceEquals(null, x)
+        if hasNulls then () else
+
+        let encoded = eventCodec.Encode(None,value)
         let e : Store.Unfold =
             {   i = 42L
                 c = encoded.EventType
-                d = encoded.Data
+                d = JsonCompressedBase64Converter.Compress encoded.Data
```

Comment:
cant this conditionality be driven by a `(compress : bool)` arg on the Property ?

Reply:
Yep, good point. Changed.